README.md

```diff
@@ -1,16 +1,129 @@
+create a `class` which is a _child_ of another `class`. 
+
+
+```js
+class Pet {
+  constructor(name) {
+    this.name = name;
+  }
+}
+
+// Inherits from Pet
+class Dog extends Pet {
+  speak() {
+    return `${this.name} says woof!`
+  }
+}
+
+let dog = new Dog("Shadow");
+
+dog.speak(); // Shadow says woof!
+```
+
+## Use the `super` Method
+
+In our code, we have 2 JavaScript `class`es: `Pet` and `Dog`. The `Dog` class
+is a _subclass_ or _child_ `class` of `Pet` and it uses the `extends` keyword
+to inherit methods from the parent `class`. With `super`, we can inherit
+functionality from the parent `class` in two different ways:
+
+```js
+class Pet {
+  constructor(name) {
+    this.name = name;
+  }
+
+  speak() {
+    return `${this.name} speaks!`
+  }
+}
+
+// Inherits from Pet
+class Dog extends Pet {
+  constructor(name) {
+    super(name)
+  }
+  speak() {
+    return `${super.speak()} ${this.name} says woof!`;
+  }
+}
+
+let dog = new Dog("Spot");
+
+dog.speak(); // Spot speaks! Spot says woof!
+```
+In `Dog`s `constructor`, `super` is used as a `method`. Whereas, `Dog`'s `speak()`
+method used `super` as an `object`. When the `super` keyword is used as method, it
+calls the parent `class` `Pet` with the parameters passed to `Dog`. This is to ensure
+that `Dog` is an instance of `Pet`. When `super` is used as an `object` that refers to
+a `Pet` instance, it can calls methods of the parent `class` `Pet` explicitly.
+
+## Explain When `class` Inheritance Should Be Used in JavaScript
+
+JavaScript can be a bit confusing since it is a dynamic language that uses syntactical
+sugar to leverage `class` implementations and inheritance similar to classic programming
+languages. However, JavaScript's inheritance model is quite powerful and fairly trivial
+to build on top of like when using a classic programming language model. As the popular
+phrase goes: "With power comes great responsibility." It can be tempting to overuse these
+features. When creating "child" `class`es, we defer to _subtyping_ relation principles
+called _(strong) behavioral subtyping_ in programming language theory. 
+
+- You probably won't use inheritance often, but as your code bases get larger, you are more
+likely to find a need for it. If you're creating a number of objects that have similar
+features, then creating a generic object type to contain all the shared functionality,
+and inheriting those features in more specialized objects can be convenient and useful.
+
+- When creating a parent-child `class` relation, it is _semantic_ rather than merely
+_syntactic_. Let's use `Vehicle` as a parent `class` with `Car` as a child
+`class` of `Vehicle` as an example. In _classical_ inheritance, it creates a _copy_ of the
+behavior from parent. However, in JavaScript when we create the object it does not copy the
+properties or behavior, it creates a _link_. `class`es should _never_ be extended unless it
```

Comment:
I think this needs clarity on the difference. Does creating a link provide some disadvantage that creating a copy prevents? Does this mean that an extended class method could potentially alter the parent's method of the same name?

This is the prototype chain, yes? I guess we could mention it here briefly if needed. It could be an aside after link (using the > in markdown) explaining how each object in JS has a prototype.

Reply:
'`class`es should _never_ be extended' I'm all for this message, but I think it needs to be clearer why never and not 'rarely'